### DISCUSSION TITLE
Create a native scroll window without using a ListView?

### DISCUSSION THREAD

**Discussion by jrmoulton** [slint-ui/slint#711] [Q&A] [answered]
In discussion #680 it was said that the API that does the scrolling is ```Flickable```. But if I create a ```Flickable``` instance, even one where the content overflows the window height, it does not create a scrollable window.

I'm trying to create a window of scrollable content where I have full control over the positioning of elements but using a ScrollView constrains the positioning of elements. How could I do this?

**Comment by ogoffart** [comment]
That should normally work. If you have a Flickable and have an element inside that is bigger, it should able to flick with the mouse, or with the mouse wheel. It will not have scroll bar though.

Maybe you want to use the ScrollArea widget, which has scrollbar?

Could you paste what you try that did not work?

**Comment by jrmoulton** [comment, marked as answer]
```
window := Window {
    min-width: 300pt;
    min-height: 388pt;
    preferred-width: 425pt;
    preferred-height: 550pt;
    property <[TextElement]> TextElements: [];
    document := ScrollView {  
        y: 10px;
        x: 10pt;
        for text_element[i] in TextElements : Text {
            y: i * 25px;
            width: parent.width - 30pt;
            wrap: word-wrap;
            text: text_element.text;
        }
    }
}
```
Using a ScrollView has the same issue as a Flickable. The content is rendered and if I manually resize the window I can see the content but there is no scroll bar. 
Is it not working because this is a for loop of elements?

**Reply by ogoffart** [reply]
I think i see the problem: Your elements are not inside of a layout, so the layout system cannot compute the size of the viewport.
You can try setting the size of the viewport explicitly with viewport-height and viewport-width property.
Or better you should use a VerticalLayout for the items.

**Reply by ogoffart** [reply]
Actually, i believe we should try to compute the size of the viewport automatically, but we don't do it for `for` element: https://github.com/sixtyfpsui/sixtyfps/issues/407

**Reply by jrmoulton** [reply]
Ahh perfect. Setting the viewport width and height on a ScrollView is giving me a scrollable that lets me control the placing of the elements. I was able to compute the size of the viewport based on the number of elements in the array. This is exactly what I was looking for! Thanks!

I could update the documentation on ScrollView and Flickable and make note in the docs that for now it is necessary to manually computer and set the viewport-width and height in order to have the scrolling properties work correctly. If that sounds ok should I also link that issue #407 in the docs?

**Reply by ogoffart** [reply]
Yes, that would be great
